lnwire/accept_channel.go

```diff
@@ -113,6 +118,7 @@ func (a *AcceptChannel) Encode(w io.Writer, pver uint32) error {
 		a.DelayedPaymentPoint,
 		a.HtlcPoint,
 		a.FirstCommitmentPoint,
+		a.UpfrontShutdownScript,
```

Comment:
add conditional logic to encode here, rather than in the codec?

Reply:
I'll separate it out as in `Decode`, but if by codec you mean `channeldb/codec` (I'm not sure what codec is :/) then won't we still need conditional logic because `WriteElement` isn't being passed a wire message? It will be passed another struct which has the field with a nil value.